core/src/schema/content/number.rs

```diff
@@ -66,224 +66,266 @@ macro_rules! number_content {
+            #[serde(deny_unknown_fields)]
+            pub enum $as {
+                $($variant($variant_ty),)*
             }
-        }
-        )*
-
-		impl From<$as> for NumberContent {
-		    fn from(value: $as) -> Self {
-			Self::$as(value)
-		    }
-		}
-	    )*
-	}
-
-	#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Default)]
-	#[serde(deny_unknown_fields)]
-    pub struct Id {
-        #[serde(default)]
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub start_at: Option<u64>,
-    }
 
-	impl NumberContent {
-	    pub fn accepts(&self, number: &Number) -> Result<()> {
-		if self.is_u64() && number.is_u64()
-                    || self.is_i64() && number.is_i64()
-                    || self.is_f64() && number.is_f64()
-                {
-                    Ok(())
-                } else {
-		    // TODO: better error
-                    Err(failed!(target: Release, "numerical type mismatch"))
+            $(
+            impl From<$variant_ty> for $as {
+                fn from(value: $variant_ty) -> Self {
+                    Self::$variant(value)
                 }
-	    }
+            }
+            )*
 
-	    pub fn kind(&self) -> &'static str {
-		match self {
-		    $(
-			$(
-			    Self::$as(number_content::$as::$variant(_)) => {
-				concat!(stringify!($as), "::", stringify!($variant))
-			    },
-			)*
-		    )*
-		}
-	    }
+            impl From<$as> for NumberContent {
+                fn from(value: $as) -> Self {
+                    Self::$as(value)
+                }
+            }
+            )*
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Default)]
+#[serde(deny_unknown_fields)]
+pub struct Id {
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub start_at: Option<u64>,
+}
 
-	    $(
-		pub fn $def() -> Self {
-		    Self::$as(number_content::$as::Range(RangeStep::default()))
-		}
-
-		pub fn $is(&self) -> bool {
-		    match self {
-			Self::$as(_) => true,
-			_ => false
-		    }
-		}
-	    )*
-	}
+impl NumberContent {
+    pub fn accepts(&self, number: &Number) -> Result<()> {
+        if self.is_u64() && number.is_u64()
+            || self.is_i64() && number.is_i64()
+            || self.is_f64() && number.is_f64()
+        {
+            Ok(())
+        } else {
+            // TODO: better error
+            Err(failed!(target: Release, "numerical type mismatch"))
+        }
     }
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq)]
 #[serde(deny_unknown_fields)]
+#[serde(default)]
 pub struct RangeStep<N> {
-    pub low: N,
-    pub high: N,
-    pub step: N,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub low: Option<N>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub high: Option<N>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub step: Option<N>,
+    #[serde(skip_serializing_if = "std::clone::Clone::clone")]
```

Comment:
What is happening here? Why skip serializing if `clone`?

Reply:
This is equivalent to testing the value to be true.